app/code/Magento/ImportExport/Test/Unit/Model/Import/Source/FileParser/FakeObjectManager.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\ImportExport\Test\Unit\Model\Import\Source\FileParser;
+
+use Magento\Framework\ObjectManagerInterface;
+
+class FakeObjectManager implements ObjectManagerInterface
+{
+    private $instances = [];
+
+    public function create($type, array $arguments = [])
+    {
+        $arguments = $this->resolveConstructorArguments($type, $arguments);
+        return new $type(...$arguments);
+    }
+
+    public function get($type)
+    {
+        if (!isset($this->instances[$type])) {
+            $this->instances[$type] = $this->create($type);
+        }
+
+        return $this->instances[$type];
+    }
+
+    public function configure(array $configuration)
+    {
+        // Fake object manager is not configurable
+    }
+
+    private function resolveConstructorArguments($type, array $arguments)
+    {
+        $constructorSignature = new \ReflectionMethod($type, '__construct');
+        $resolvedArguments = [];
+        foreach ($constructorSignature->getParameters() as $parameter) {
+            $arguments = $this->assertParameterValue($arguments, $parameter);
+
+            $resolvedArguments[] = $arguments[$parameter->getName()] ?? $parameter->getDefaultValue();
+        }
+
+        return $resolvedArguments;
```

Comment:
when we provide an implementation of Fake objects, who and how will test this implementation? 

For example, current function is **10 lines of code**. How can we make sure that this code is safe to rely upon?